/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

// Defines protos for storing experiment data inside Summary tags.

syntax = "proto3";

import "tensorboard/plugins/hparams/api.proto";

package tensorboard.plugins.hparams;

message SessionStartSummary {
  // The directory where checkpoints are saved.
  string checkpoint_path = 1;

  // An optional link to a website monitoring the session.
  string monitor_url = 2;

  // The name of the session group containig this one. If empty sessions are
  // essentially not grouped -- each group contains exactly one session.
  string group_name = 3;

  repeated HParamVal hparam_vals = 4;
}

message SessionEndSummary {
  Status status = 1;
}
